src/pages/api/cities.js

```diff
@@ -0,0 +1,84 @@
+import AppDataSource from '@/data-source';
+import User from '@/entity/User';
+import City from '@/entity/City';
+
+export default async function handler(req, res) {
+  if (!AppDataSource.isInitialized) {
+    await AppDataSource.initialize();
+  }
+  const userRepo = AppDataSource.getRepository(User);
+  const cityRepo = AppDataSource.getRepository(City);
+
+  let cityIsAlreadySavedByCurrentUser;
+
+  if (req.method === 'POST') {
```

Comment:
I think you should extract the code inside if blocks in separate functions. You can let the functions in this file. The if statements should look something like this:
if (req.method === 'POST') {
  createCity()
}
...
I think the code will look cleaner.